rbd-target-api.py

```diff
@@ -2816,13 +2817,46 @@ def run(self):
                     config.refresh()
 
 
+def get_ssl_files_from_mon():
+    client_name = settings.config.cluster_client_name
+    temp_files = []
+    with rados.Rados(conffile=settings.config.cephconf,
+                     name=client_name) as cluster:
+        cmd = {"prefix": "mon config-key get",
+               "key": "iscsi/{}/iscsi-gateway.crt".format(client_name)}
+        ret, crt_data, outs = cluster.mon_command(json.dumps(cmd), b'')
+        if not ret:
+            return temp_files
+
+        cmd["key"] = "iscsi/{}/iscsi-gateway.key".format(client_name)
```

Comment:
I would need to double-check from the `cephadm`-side, but for security reasons, is the private key locked down so only the iSCSI user can read them?

Reply:
I don't think it's limited to the iscsi user unfortunately. Though you would need a keyring with I guess mon read caps.. I haven't tested this. So you'd at least need to be authenticated.

But a very interesting point

Reply:
using mon caps you could limit access to this prefix for just the iscsi user though. We use similar things for the mgr, for instance, although we achieve that in particular by hardcoding those caps via the mgr profile.

In https://docs.ceph.com/docs/master/rados/operations/user-management/#authorization-capabilities you will find a few examples of this, but I would imagine you'd want to limit access to that key via

> mon 'allow command "config-key get" with key prefix iscsi/$client_name/iscsi-gateway.key,allow command "config-key set" with key prefix iscsi/$client_name/iscsi-gateway.key"

If you want to see a lot more examples of how this could be achieved, you can take a look at `qa/workunits/mon/test_config_key_caps.sh`.